src/game.cpp

```diff
@@ -2569,7 +2569,17 @@ void Game::processUserInput(VolatileRunFlags *flags,
 			|| noMenuActive() == false
 			|| guienv->hasFocus(gui_chat_console)) {
 		input->clear();
+#ifdef HAVE_TOUCHSCREENGUI
+		g_touchscreengui->Hide();
```

Comment:
Touchscreen gui is shown uppon exiting menu. But on entering due to timing effects it may not be hidden until first touch event occurs. It's obviously not necessary to show the gui while menu is active so this fixes the glitch.